guidance/models/_azure_guidance.py

```diff
@@ -0,0 +1,167 @@
+import requests
+import os
+import base64
+import json
+import urllib.parse
+from ._model import Engine, Model, EngineCallResponse
+from .._chat import Phi3ChatTemplate
+from ._byte_tokenizer import ByteTokenizer
+
+
+class AzureGuidanceEngine(Engine):
+    """This connects to a remote guidance server on Azure and runs all computation using the remote engine."""
+
+    def __init__(self, server_url, max_streaming_tokens=1000, chat_template=None):
+        if (
+            server_url is None
+            or isinstance(server_url, str)
+            and len(server_url.strip()) == 0
+        ):
+            server_url = os.getenv("AZURE_GUIDANCE_URL", "")
+        elif not isinstance(server_url, str):
+            raise ValueError("server_url must contain a URL string.")
+
+        if not server_url.startswith("http"):
```

Comment:
I suppose we can't enforce https, but should we at least issue a warning?

Reply:
Good point!  I changed it to "https://", which all azure endpoints will require, I think.

Reply:
anything against adding http:// ? or at least http://localhost: and http://127.0.0.1: ?